RecoHGCal/TICL/plugins/TrackstersProducer.cc

```diff
@@ -0,0 +1,103 @@
+// Author: Felice Pantaleo,Marco Rovere - felice.pantaleo@cern.ch,marco.rovere@cern.ch
+// Date: 09/2018
+// Copyright CERN
+
+// user include files
+#include <vector>
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+#include "DataFormats/CaloRecHit/interface/CaloCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFCluster.h"
+
+#include "RecoHGCal/TICL/interface/PatternRecognitionAlgoBase.h"
+#include "RecoHGCal/TICL/interface/Trackster.h"
+#include "PatternRecognitionbyCA.h"
+#include "PatternRecognitionbyMultiClusters.h"
+
+class TrackstersProducer : public edm::stream::EDProducer<> {
+ public:
+  TrackstersProducer(const edm::ParameterSet &);
+  ~TrackstersProducer() override {}
+  static void fillDescriptions(edm::ConfigurationDescriptions &descriptions);
+
+  void produce(edm::Event &, const edm::EventSetup &) override;
+
+ private:
+  edm::EDGetTokenT<std::vector<reco::CaloCluster>> clusters_token_;
+  edm::EDGetTokenT<std::vector<std::pair<unsigned int, float>>> filtered_layerclusters_mask_token_;
+  edm::EDGetTokenT<std::vector<float>> original_layerclusters_mask_token_;
+
+  std::unique_ptr<PatternRecognitionAlgoBase> myAlgo_;
+};
+DEFINE_FWK_MODULE(TrackstersProducer);
+
+TrackstersProducer::TrackstersProducer(const edm::ParameterSet& ps)
+    : myAlgo_(std::make_unique<PatternRecognitionbyCA>(ps)) {
```

Comment:
is there a plan to make this concrete PR algo selection configurable?
If no, the producer name is inappropriate and better have ```ByCA``` in the name.